### PR TITLE
[3.2] docs(dialogs): Revamped Dialog Overview

### PR DESCRIPTION
**Pull request by xristianstefanov**

### CODE REVIEW

File: components/dialog/overview.md [hunk 1]
```diff
@@ -12,15 +12,17 @@ position: 0
...
 
-1. Add the `TelerikDialog` tag.
-1. Set the `Visible` parameter via one-way or two-way binding.
-1. Set a `Title`.
+2. Set the `Visible` parameter to a `bool` object. It supports one-way and two-way binding.
 
-The following example demonstrates how to set up the Dialog with its default configuration.
+3. Set the `Title` property to a `string` object.
+
+4. Define the `DialogContent` inside the `TelerikDialog` tag.
```
Comment (dimodi) [inline components/dialog/overview.md:23]: It's not clear if this is a parameter, a child tag, or can be both.

```suggestion
4. Define the `DialogContent` inside the `TelerikDialog` tag.
```

File: components/dialog/overview.md [hunk 2]
```diff
@@ -45,38 +47,108 @@ The following example demonstrates how to set up the Dialog with its default con
 }
 ````
 
->caption The result from the above code snippet.
+## Predefined Dialogs
+
+Predefined Dialogs are styled substitutes to the standard confirm, alert and prompt dialogs. They are separate from the main Dialog component. [Read more about the Blazor Predefined Dialogs]({%slug dialog-predefined%}).
```
Comment (dimodi) [inline components/dialog/overview.md:52]: What do you want to say with "They are separate from the main Dialog component." - what should the reader do or learn from this information?

```suggestion
Predefined Dialogs are styled substitutes to the standard browser dialogs - confirm, alert and prompt. They are separate from the main Dialog component. [Read more about the Blazor Predefined Dialogs]({%slug dialog-predefined%}).
```

File: components/dialog/overview.md [hunk 3]
```diff
@@ -45,38 +47,108 @@ The following example demonstrates how to set up the Dialog with its default con
...
+
+## Header
+
+The Dialog provides header customization. [Read more about the Dialog Header]({%slug dialog-header%}).
```
Comment (dimodi) [inline components/dialog/overview.md:56]: Briefly mention something valuable here, for example the developer can customize the title and toggle the close button.

```suggestion
The Dialog provides header customization. [Read more about the Dialog Header]({%slug dialog-header%}).
```

File: components/dialog/overview.md [hunk 4]
```diff
@@ -45,38 +47,108 @@ The following example demonstrates how to set up the Dialog with its default con
...
+
+## Action Buttons
+
+The Dialog provides options for rendering action buttons and customizing their text. [Read more about the Dialog Action Buttons]({%slug dialog-action-buttons%}).
```
Comment (dimodi) [inline components/dialog/overview.md:60]:
```suggestion
The Dialog provides options for rendering action buttons and customizing their text and layout. [Read more about the Dialog Action Buttons]({%slug dialog-action-buttons%}).
```

File: components/dialog/overview.md [hunk 5]
```diff
@@ -45,38 +47,108 @@ The following example demonstrates how to set up the Dialog with its default con
...
 
-![](images/dialog-basic-configuration.png)
+## Events
 
-## Features
+The Blazor Dialog fires visible change events that you can handle and further customize its behavior. [Read more about the Blazor Dialog events]({%slug dialog-events%}).
```
Comment (dimodi) [inline components/dialog/overview.md:64]: Do not use code as plain text, as it makes the sentence hard to understand.

```suggestion
The Blazor Dialog fires a `VisibleChanged` event to customize the application behavior and respond to user actions. [Read more about the Blazor Dialog events]({%slug dialog-events%}).
```

File: components/dialog/overview.md [hunk 6]
```diff
@@ -45,38 +47,108 @@ The following example demonstrates how to set up the Dialog with its default con
...
+    article style + table {
+        table-layout: auto;
+        word-break: normal;
+    }
+</style>
+| Parameter | Type and Default Value | Description |
+| --- | --- | --- |
+| `ButtonsLayout` | `DialogButtonsLayout` enum <br /> (`Stretched`) | Defines the layout of the actions button in the footer. See more in the [Action Buttons article]({%slug  dialog-action-buttons%})). |
+| `Class` | `string` | Renders a custom CSS class to the `<div class="k-window k-dialog">` element. |
+| `CloseOnOverlayClick` | `bool` | Defines whether clicking on the modal overlay should close the Dialog. |
```
Comment (dimodi) [inline components/dialog/overview.md:80]: "whether" is more correct, but native speakers say the "if" is OK too. And will make the description shorter.

```suggestion
| `CloseOnOverlayClick` | `bool` | Defines if clicking on the modal overlay should close the Dialog. |
```

File: components/dialog/overview.md [hunk 7]
```diff
@@ -45,38 +47,108 @@ The following example demonstrates how to set up the Dialog with its default con
...
+| `FocusedElementSelector` | `string` | Defines the CSS selector of the initially focused item on open. By default, it is the first focusable item in the dialog. |
+| `Height` | `string` | Defines the height of the Dialog. |
+| `ShowCloseButton` | `bool` <br /> (`true`) | Defines the close behavior of the component - whether the component should render close flat button in the titlebar. See more in the [Header article]({%slug  dialog-header%}). |
```
Comment (dimodi) [inline components/dialog/overview.md:83]: "close behavior" means something different

```suggestion
| `ShowCloseButton` | `bool` <br /> (`true`) | Defines if the component should render a Close button in the titlebar. See more in the [Header article]({%slug  dialog-header%}). |
```

File: components/dialog/overview.md [hunk 8]
```diff
@@ -45,38 +47,108 @@ The following example demonstrates how to set up the Dialog with its default con
...
+| `Title` | `string` | Defines the Dialog title. |
+| `Visible` | `bool` | Defines the Dialog visibility. |
```
Comment (dimodi) [inline components/dialog/overview.md:85]: Visibility and rendering are different things. Which one is it?

```suggestion
| `Visible` | `bool` | Defines the Dialog visibility. |
```

File: components/dialog/overview.md [hunk 9]
```diff
@@ -45,38 +47,108 @@ The following example demonstrates how to set up the Dialog with its default con
...
+</style>
+| Parameter | Type and Default Value | Description |
+| --- | --- | --- |
+| `ButtonsLayout` | `DialogButtonsLayout` enum <br /> (`Stretched`) | Defines the layout of the actions button in the footer. See more in the [Action Buttons article]({%slug  dialog-action-buttons%})). |
+| `Class` | `string` | Renders a custom CSS class to the `<div class="k-window k-dialog">` element. |
+| `CloseOnOverlayClick` | `bool` | Defines whether clicking on the modal overlay should close the Dialog. |
+| `FocusedElementSelector` | `string` | Defines the CSS selector of the initially focused item on open. By default, it is the first focusable item in the dialog. |
+| `Height` | `string` | Defines the height of the Dialog. |
+| `ShowCloseButton` | `bool` <br /> (`true`) | Defines the close behavior of the component - whether the component should render close flat button in the titlebar. See more in the [Header article]({%slug  dialog-header%}). |
+| `Title` | `string` | Defines the Dialog title. |
```
Comment (dimodi) [inline components/dialog/overview.md:84]:
```suggestion
| `Title` | `string` | Sets the Dialog title. |
```

File: components/dialog/overview.md [hunk 10]
```diff
@@ -45,38 +47,108 @@ The following example demonstrates how to set up the Dialog with its default con
...
+        word-break: normal;
+    }
+</style>
+| Parameter | Type and Default Value | Description |
+| --- | --- | --- |
+| `ButtonsLayout` | `DialogButtonsLayout` enum <br /> (`Stretched`) | Defines the layout of the actions button in the footer. See more in the [Action Buttons article]({%slug  dialog-action-buttons%})). |
+| `Class` | `string` | Renders a custom CSS class to the `<div class="k-window k-dialog">` element. |
+| `CloseOnOverlayClick` | `bool` | Defines whether clicking on the modal overlay should close the Dialog. |
+| `FocusedElementSelector` | `string` | Defines the CSS selector of the initially focused item on open. By default, it is the first focusable item in the dialog. |
+| `Height` | `string` | Defines the height of the Dialog. |
```
Comment (dimodi) [inline components/dialog/overview.md:82]:
```suggestion
| `Height` | `string` | Sets the height of the Dialog. |
```

File: components/dialog/overview.md [hunk 11]
```diff
@@ -45,38 +47,108 @@ The following example demonstrates how to set up the Dialog with its default con
...
+| `ShowCloseButton` | `bool` <br /> (`true`) | Defines the close behavior of the component - whether the component should render close flat button in the titlebar. See more in the [Header article]({%slug  dialog-header%}). |
+| `Title` | `string` | Defines the Dialog title. |
+| `Visible` | `bool` | Defines the Dialog visibility. |
+| `Width` | `string` | Defines the width of the Dialog. |
```
Comment (dimodi) [inline components/dialog/overview.md:86]:
```suggestion
| `Width` | `string` | Sets the width of the Dialog. |
```

**Review by dimodi — CHANGES_REQUESTED** [review]
Follow the inline comments.

File: components/dialog/overview.md (hunk 1 above)
Comment (jivanova) [reply components/dialog/overview.md:23]: both - maybe "Set the dialog content through the `DialogContent` RenderFragment parameter" will be explanatory

**Review by dimodi — APPROVED** [review]

**Pull request merged**